arch/arm64/src/Toolchain.defs

```diff
@@ -96,6 +100,10 @@ ifeq ($(CONFIG_ARCH_FPU),y)
 endif
 
 ARCHCFLAGS += -fno-common
+
+ifeq ($(CONFIG_IMX9_BOOTLOADER),y)
```

Comment:
I think this would fit better into Make.defs either under arch/arm64/src/imx9 or under boards/.../ ?  If board/chip specific options get added into Toolchain.defs, it will soon be quite full of those